zerver/actions/user_topics.py

```diff
@@ -6,73 +6,58 @@
 
 from zerver.lib.exceptions import JsonableError
 from zerver.lib.timestamp import datetime_to_timestamp
-from zerver.lib.user_topics import add_topic_mute, get_topic_mutes, remove_topic_mute
+from zerver.lib.user_topics import get_topic_mutes, set_user_topic_visibility_policy
 from zerver.models import Stream, UserProfile, UserTopic
 from zerver.tornado.django_api import send_event
 
 
-def do_mute_topic(
+def do_set_user_topic_visibility_policy(
     user_profile: UserProfile,
     stream: Stream,
     topic: str,
-    date_muted: Optional[datetime.datetime] = None,
+    *,
+    visibility_policy: int,
+    last_updated: Optional[datetime.datetime] = None,
     ignore_duplicate: bool = False,
+    skip_muted_topics_event: bool = False,
 ) -> None:
-    if date_muted is None:
-        date_muted = timezone_now()
-    assert stream.recipient_id is not None
-    add_topic_mute(
-        user_profile,
-        stream.id,
-        stream.recipient_id,
-        topic,
-        date_muted,
-        ignore_duplicate=ignore_duplicate,
-    )
-
-    # This first muted_topics event is deprecated and will be removed
-    # once clients are migrated to handle the user_topic event type
-    # instead.
-    muted_topics_event = dict(type="muted_topics", muted_topics=get_topic_mutes(user_profile))
-    send_event(user_profile.realm, muted_topics_event, [user_profile.id])
-
-    user_topic_event: Dict[str, Any] = {
-        "type": "user_topic",
-        "stream_id": stream.id,
-        "topic_name": topic,
-        "last_updated": datetime_to_timestamp(date_muted),
-        "visibility_policy": UserTopic.MUTED,
-    }
-
-    send_event(user_profile.realm, user_topic_event, [user_profile.id])
-
-
-def do_unmute_topic(
-    user_profile: UserProfile, stream: Stream, topic: str, *, skip_muted_topics_event: bool = False
-) -> None:
-    # Note: If you add any new code to this function, the
-    # remove_topic_mute call in do_update_message will need to be
-    # updated for correctness.
-    try:
-        remove_topic_mute(user_profile, stream.id, topic)
-    except UserTopic.DoesNotExist:
-        raise JsonableError(_("Topic is not muted"))
-
+    if last_updated is None:
+        last_updated = timezone_now()
+
+    if visibility_policy == UserTopic.VISIBILITY_POLICY_INHERIT:
+        # Note: If you add any new code to this block, the
+        # set_user_topic_visibility_policy call in do_update_message will need to be
+        # updated for correctness.
```

Comment:
I think this comment is stale and can be deleted; bd04733c0f5a2fe8cbebccc515847a2e96da559c added the  `skip_muted_topics_event` parameter that replaced that code duplication.